### ISSUE TITLE
dataStor reading help

### ISSUE THREAD

**Issue by TheGreenDragon** [gtc309/otr#5]
I have put a quick mock up of a CSV reader, but I need some help. Since I am using a variable in the CSV file creation (the truck number) the load file screen needs to give me a truck number to access the correct file. i used the generic variable truckNumber for this for now.

**Comment by gtc309** [comment]
That is a good start. Your reader can be built to serve a dual role. One
is for the login, it can return (from the password file) a login match
or fail and second it can grab all the data for a truck. I think what is
needed is a lookup button on the main panel which calls your reader then
you return a list of truck numbers for the person to select from. Once
selected control returns to you and you return the data for the specific
truck. However we would also want to have a selectable date rang in as
well therefore we only grab data appropriate for the current users
context. 
What is needed
A button to look up
A list returned with truck numbers and dates
A selection button for an item on the list and a multi select for an
aggregate of items (this will also requre another screen where we can
see a list of history in spreadsheet type format). 
First step, lookup button, and return of dates and truck numbers.

Taking a step back we need to add a date field to the array which you a
re writing. I will add that in today.

**Comment by TheGreenDragon** [comment]
I have pushed an update to read the date from the array and write it to the CSV. However, I did notice there does not seem to be the same level of error correction on the date field in the main form. It allowed me to clear the field and tab out leaving the field blank on the form with no error message. It will also allow the data to be saved with no date information present.

**Comment by gtc309** [comment]
Yes that part needs to be built. I just stuck a date on the array. If
someone wants to jump in and do the date validation function go for it.

**Comment by gtc309** [comment]
The date field now requires a date entry in the form of MM/DD/YY. It will not save unless a date is entered.

**Comment by TheGreenDragon** [comment]
One issue I have noticed since the date fix, when you input an invalid date or leave the field blank, you are prompted to input the date into an error window. The data input into this field, when transferred to the main window loses any leading zero's. This prevents the file from saving and the date must be manually corrected. I am looking at it now.

**Comment by TheGreenDragon** [comment]
I see in otr2.java where the call for the test is made, and I see in test.java where the date test is done and returned, but I am not sure how the leading zero is truncated. Any thoughts?

**Comment by gtc309** [comment]
The strange thing is the zeros stay when entering the date a second time
from the dialog box during a save. I am not sure why the 0 is being
stripped before then. When watching the variables values they seem to
stay in place.

**Comment by TheGreenDragon** [comment]
So my thought for the load file box has a text box for the user to manually input a directory and defaults to the user directory just like the save does. Below that a small tree window. On the right a load button. The bottom half will be a table that imports the headers and csv data from the file when it is highlighted. Upon clicking load the array will be populated with the most recent data.

Not that I have any clue how I'm going to implement this yet, but I'm googling the hell out of it! 

Should it be a new class or in one of the existing ones?

**Comment by gtc309** [comment]
You will be making a few classes. The datastor class will get a new
method for sending a array to your window that displays the spreadsheet
format of data for a trucks history. Then you will probably want to make
an entire new class and form that displays the information. Then in that
form if if a user selects a range of data or a specific data item,
another new class can be developed to handle that, if it is one item the
user would be sent over to the otr2.java form, if it is a list of items
they would be sent to a new form that gives totals based on the list and
any new entry they want to add. Swaping forms is not hard, and actually
at some point we should do a little redesign bad make one frame with
multiple panels that display at apporpriate times.

**Comment by TheGreenDragon** [comment]
Something tells me this will NOT be done by 11:59 tonight....

**Comment by gtc309** [comment]
No one step at a time. I think first is build a method  that when the
load button is clicked a window pops up with a list of truck numbers to
choose from.

**Comment by TheGreenDragon** [comment]
I have begun working on a class named loadMenu that will populate a form when "load" is pressed. I am using it as a tutorial in gui programming with IntelliJ so progress will be slow.

**Comment by gtc309** [comment]
Ok if you need me to look at something today, just post it and let me
know. I am finishing up a final defense presentation now and will pop
into the project afterwards.

**Comment by TheGreenDragon** [comment]
Small push includes new load window not completed but done for today.

**Comment by gtc309** [comment]
Can you send over your otr2.form Your button works when I run the
program but it does not show in the design window. It is because the
form is not in sync with the .java.

**Comment by gtc309** [comment]
I threw in some code from Oracle tht may help you sort out the menu. I put comments around yours so you can compare and switch things around. In the end when clicking the Load button it should cause  setVisible(true) event that pops up that form.
